src/ReadData.cpp

```diff
@@ -0,0 +1,133 @@
+#include "Data.h"
+#include "ReadData.h"
```

Comment:
can you reuse the code from `BinaryCoding.h`?

Reply:
I'm going to take a look at it.